superset/datasets/schemas.py

```diff
@@ -49,14 +49,14 @@ class DatasetColumnsPutSchema(Schema):
     column_name = fields.String(required=True, validate=Length(1, 255))
     type = fields.String(allow_none=True)
     advanced_data_type = fields.String(allow_none=True, validate=Length(1, 255))
-    verbose_name = fields.String(allow_none=True, Length=(1, 1024))
+    verbose_name = fields.String(allow_none=True, metadata={Length: (1, 1024)})
```

Comment:
I assume you're doing this because Marshmallow deprecated passing arbitrary keys as metadata, and now recommends explicitly using the `metadata` argument, ie:

```python
fields.String(allow_none=True, description="foo")  # DEPRECATED
fields.String(allow_none=True, metadata={"description": "foo"})  # RECOMMENDED
```

The reason why the latter is recommended is because the former will silently swallow keywords arguments, which is exactly what's happening here; `Length()` is supposed to be a validator, not metadata. So you want:

```suggestion
    verbose_name = fields.String(allow_none=True, validate=Length(1, 1024))
```